homeassistant/components/growatt_server/config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Config flow for growatt server integration."""
+import voluptuous as vol
+
+import growattServer
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME, CONF_USERNAME, CONF_PASSWORD
+from .const import DOMAIN, CONF_PLANT_ID, DEFAULT_NAME
+
+
+class GrowattServerConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow class."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialise growatt server flow."""
+        self.user_input = {}
+        self.api = growattServer.GrowattApi()
+
+    async def async_step_plant(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        user_input = {**self.user_input, **user_input}
+
+        if CONF_PLANT_ID not in user_input:
+            login_response = self.api.login(
+                user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+            )
+            user_id = login_response["userId"]
+            plant_info = self.api.plant_list(user_id)
+
+            if len(plant_info["data"]) == 0:
+                return await self._show_user_form({"base": "no_plants"})
```

Comment:
We should make this an abort, since there is no way we can finish it from here ?